tests/ui/closures/2229_closure_analysis/migrations/auto_traits.fixed

```diff
@@ -26,7 +26,7 @@ fn test_send_trait() {
         //~| HELP: add a dummy let to cause `fptr` to be fully captured
         *fptr.0 = 20;
         //~^ NOTE: in Rust 2018, this closure captures all of `fptr`, but in Rust 2021, it will only capture `fptr.0`
-    } });
+    } }).join().unwrap();
```

Comment:
Why is this needed? Doesn't look like a UB fix. Is this about fixing memory leaks? Is that even a goal or should Miri be run with `-Zmiri-ignore-leaks` when running ui tests?

Reply:
Without this join, the thread can execute after the function returns, which is a use-after-free.

Reply:
Oh good point!